src/Provider/TMSProvider.js

```diff
@@ -23,32 +24,56 @@ function preprocessDataLayer(layer) {
             max: 18,
         };
     }
+    layer.fx = layer.fx || 0.0;
+}
+
+function getVectorTile(tile, coords, layer) {
+    const url = URLBuilder.xyz(coords, layer);
+
+    if (layer.type == 'color') {
+        return VectorTileHelper.getVectorTileTextureByUrl(url, tile, layer, coords)
+            .then((texture) => {
+                const result = {
+                    texture,
+                };
+                return result;
+            });
```

Comment:
could not this wrapping be done in `VectorTileHelper.getVectorTileTextureByUrl` and the `URLBuilder.xyz` be done outside of the format switch (see below), so as to remove this function entirely ?

Reply:
This wrapping could, but unfortunately the url no.

Reply:
A quick update: I need to keep this method if I want to support different type of layer (like `geometry`), as it is the only way for now ;)